esp/logging/logginglib/loggingagentbase.cpp

```diff
@@ -294,8 +294,8 @@ void CLogContentFilter::filterLogContent(IEspUpdateLogRequestWrap* req)
     StringBuffer updateLogRequestXML;
     toXML(updateLogRequestTree, updateLogRequestXML);
     ESPLOG(LogMax, "filtered content and option: <%s>", updateLogRequestXML.str());
-    req->clearOriginalContent();
-    req->setUpdateLogRequest(updateLogRequestXML.str());
+
+    return new CUpdateLogRequestWrap(req->getGUID(), req->getOption(), updateLogRequestXML.str());
```

Comment:
This whole bit of code feels very heavyweight for logging...

Reply:
I am not sure that I understand the comment. Are you saying that I should not create a new CUpdateLogRequestWrap?  The filterLogContent() may be called by multiple logging agents in different threads. If each of them does not have their copy of logging request, one logging agent may change the original request and another agent may not see the original request.

Reply:
The logging function is designed to handle multiple logging agents per application. It should also handle different kinds of logging requests (xml string, ptree, and one mixed group for now). A user configures some filter to filter the requests before the requests are queued. If one logging attempt is failed, the request has to be re-queued (not in this method). A user may also configure some mapping to pick the parts of the requests to send out (not in this method). When I have time, some parts of the code do need to be optimized.